pkg/cloudprovider/openstack/openstack.go

```diff
@@ -625,28 +626,41 @@ func (lb *LoadBalancer) UpdateTCPLoadBalancer(name, region string, hosts []strin
 func (lb *LoadBalancer) DeleteTCPLoadBalancer(name, region string) error {
 	glog.V(4).Infof("DeleteTCPLoadBalancer(%v, %v)", name, region)
 
-	vip, err := getVipByName(lb.network, name)
-	if err != nil {
-		return err
+	vip, vipErr := getVipByName(lb.network, name)
+	vipExists := true
+	if vipErr != nil {
+		if vipErr != ErrNotFound {
+			return vipErr
+		}
+		vipExists = false
```

Comment:
minor nit: I'd prefer a single assignment statement as follows after the if statement, rather than both one before and one inside.  It feels a bit wierd to assign vipExists := true before we actually know whether or not it does.
Instead, you could just do this after the if statement.

vipExists = (vipErr == nil)